src/i18n/TextCollection/i18nTextCollector.php

```diff
@@ -1075,6 +1078,9 @@ public function getDefaultLocale()
 
     public function setDefaultLocale($locale)
     {
+        if ($locale && str_contains($locale, '..')) {
+            throw new InvalidArgumentException('Locale must not contain ".."');
+        }
```

Comment:
No locale has `..` in its name, so this is easy to do and protects against... well, probably not much because this can't really be run by arbitrary users. But it seems sensible anyway.